pkg/volume/csi/csi_plugin.go

```diff
@@ -576,32 +576,32 @@ func (p *csiPlugin) NewDetacher() (volume.Detacher, error) {
 
 // TODO change CanAttach to return error to propagate ability
 // to support Attachment or an error - see https://github.com/kubernetes/kubernetes/issues/74810
-func (p *csiPlugin) CanAttach(spec *volume.Spec) bool {
+func (p *csiPlugin) CanAttach(spec *volume.Spec) (bool, error) {
 	driverMode, err := p.getDriverMode(spec)
 	if err != nil {
-		return false
+		return false, err
```

Comment:
Can we add some unit tests that actually trigger an error?

Reply:
@msau42 added tests.